File: app/src/main/java/org/helllabs/android/xmp/modarchive/result/RandomResult.java
package org.helllabs.android.xmp.modarchive.result;

import org.helllabs.android.xmp.R;
import org.helllabs.android.xmp.modarchive.request.ModuleRequest;

import android.os.Bundle;

public class RandomResult extends ModuleResult implements ModuleRequest.OnResponseListener {
    @Override
    public void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle(R.string.search_random_title);
    }

    @Override
    protected void makeRequest(final String query) {
        final ModuleRequest request = new ModuleRequest(apiKey, ModuleRequest.RANDOM);
        request.setOnResponseListener(this).send();
    }
}

File: app/src/main/java/org/helllabs/android/xmp/modarchive/result/RandomResult.kt
package org.helllabs.android.xmp.modarchive.result

import android.os.Bundle
import org.helllabs.android.xmp.R
import org.helllabs.android.xmp.modarchive.request.ModArchiveRequest
import org.helllabs.android.xmp.modarchive.request.ModArchiveRequest.OnResponseListener
import org.helllabs.android.xmp.modarchive.request.ModuleRequest

class RandomResult : ModuleResult(), OnResponseListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTitle(R.string.search_random_title)
    }

    override fun makeRequest(query: String?) {
        val request = ModuleRequest(apiKey, ModArchiveRequest.RANDOM)
        request.setOnResponseListener(this).send()
    }
}
